### PR TITLE
Improve Cobigen reliese script

### PR DESCRIPTION
**Pull request by sanjaykumarcg**
DEVC-78.

i have dome the changes inside git_repo file for init_submodule method .

@devonfw/cobigen

### CODE REVIEW

File: scripts/src/tools/git_repo.py [hunk 1]
```diff
@@ -61,8 +62,8 @@ def update_and_clean(self):
 
     def checkout(self, branch_name):
         log_info("Checkout " + branch_name)
-        self.__repo.git.checkout(branch_name)
-        self.update_and_clean()
+        self.__repo.git.checkout(branch_name)        
+        #self.update_and_clean()
```
Comment (jdiazgon) [inline scripts/src/tools/git_repo.py]: I don't exactly remember if we commented the `update_and_clean()` for any special reason, but if not, can you check if the script stills works with this line uncommented?

I think it is a good practice to call this method each time there is a checkout.

File: scripts/src/tools/git_repo.py [hunk 2]
```diff
@@ -135,17 +136,14 @@ def merge(self, source: str, target: str) -> None:
                 self.__repo.git.execute("git merge --abort")
                 self.reset()
                 sys.exit()
-    def init_submodule(self, submodule_path: str) -> None:               
-        self.__repo.git.submodule("init") 
-        self.__repo.git.execute("git submodule init")        
-        self.__repo.git.submodule("update")
-                         
+    def init_submodule(self, submodule_path: str) -> None:
```
Comment (jdiazgon) [inline scripts/src/tools/git_repo.py]: Please remove this method as it is not needed now.

**Comment by jdiazgon** [comment]
Please also remove the last 3 commits, as you have [added and modified](https://github.com/devonfw/tools-cobigen/pull/818/commits/50a95e8700f907407d2f57613e8c2851a8b1807f) the submodules like tools-cobigen submodule and devonfw-docgen submodule, but this was not intended.

When the changes are tested I will merge.

File: scripts/src/tools/git_repo.py (hunk 1 above)
Comment (sanjaykumarcg) [reply scripts/src/tools/git_repo.py]: okay,I am checking with uncommented and let you know the result .

Comment (sanjaykumarcg) [reply scripts/src/tools/git_repo.py]: After uncomment it is still working fine .

**Pull request merged**